slack-lint/src/main/java/slack/lint/text/SpanPointMarkDangerousCheckDetector.kt

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (C) 2021 Slack Technologies, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package slack.lint.text
+
+import com.android.tools.lint.client.api.UElementHandler
+import com.android.tools.lint.detector.api.Category
+import com.android.tools.lint.detector.api.Detector
+import com.android.tools.lint.detector.api.Implementation
+import com.android.tools.lint.detector.api.Issue
+import com.android.tools.lint.detector.api.JavaContext
+import com.android.tools.lint.detector.api.LintFix
+import com.android.tools.lint.detector.api.Severity
+import com.android.tools.lint.detector.api.SourceCodeScanner
+import com.android.tools.lint.detector.api.UastLintUtils
+import org.jetbrains.uast.UBinaryExpression
+import org.jetbrains.uast.UExpression
+import org.jetbrains.uast.UReferenceExpression
+import org.jetbrains.uast.UastBinaryOperator
+import org.jetbrains.uast.tryResolve
+import slack.lint.text.SpanPointMarkDangerousCheckDetector.Companion.ISSUE
+import slack.lint.util.sourceImplementation
+
+/**
+ * Checks for SpanPointMarkDangerousCheck. See [ISSUE].
+ */
+class SpanPointMarkDangerousCheckDetector : Detector(), SourceCodeScanner {
+
+  companion object {
+    private fun Implementation.toIssue() = Issue.create(
+      id = "SpanPointMarkDangerousCheck",
+      briefDescription = "Check Span Flags Using Bitmask",
+      explanation =
+      "Spans flags can have priority or other bits set. " +
+        "Ensure that Span flags are checked using " +
+        "`currentFlag and Spanned.SPAN_POINT_MARK_MASK == desiredFlag` " +
+        "rather than just `currentFlag == desiredFlag`",
```

Comment:
Can we just use a raw string? You can multiline the string (while keeping a flowing single line) by adding `\` to the end of the line

Reply:
In my testing \ did not work. Is this specific to `explanation` field?

```kotlin
@Test
  fun `raw string - when contains backslash - removes backslash`() {
    val underTest = """
        This \
        is the same line
    """.trimIndent()
    TestCase.assertEquals("This is the same line", underTest)
  }
```
```
expected:<This []is the same line> but was:<This [\
]is the same line>
```

Reply:
It is part of lint handling of strings, not kotlin trimIndent() strings

Reply:
👍 https://github.com/slackhq/slack-lints/pull/14/commits/dde29a525cbba14ecd315a20b6d2ac488c7a8f5d